android/src/main/java/com/bitmovin/player/reactnative/extensions/NativeCollections.kt

```diff
@@ -0,0 +1,18 @@
+package com.bitmovin.player.reactnative.extensions
+
+import com.facebook.react.bridge.ReadableArray
+import com.facebook.react.bridge.ReadableMap
+
+inline fun <reified T> ReadableArray.toList(): List<T?> = (0 until size()).map { i ->
+    getDynamic(i).let {
+        when (T::class) {
+            Boolean::class -> it.asBoolean() as T
+            String::class -> it.asString() as T
+            Double::class -> it.asDouble() as T
+            Int::class -> it.asInt() as T
+            ReadableArray::class -> it.asArray() as T
+            ReadableMap::class -> it.asMap() as T
+            else -> null
```

Comment:
I would throw an exception in this case.
`IllegalArgumentException` maybe and stating that the input type is not supported. 
(I think) then we can also drop the nullability of the entries in the return list.

Reply:
Btw. By convention the file would be named after the Type we are extending i.e. `ReadableArrayExtensions`

Reply:
I don't think it's a good idea to throw an exception for nullable items. The idea is that `ReadableArray` represents an array from javascript. And in javascript arrays can have multiple items of different types (including `null`). So the idea is that `ReadableArray` should represent the same structure in Java/Kotlin as it does in javascript.

What we can do in this case is to keep the nullable items as optionals (what we're already doing here) or drop the nullable items when parsing the structure from `ReadableArray` to `List`. But throwing an exception doesn't make sense to me since nullable items are common in JS arrays so they shouldn't represent an error.

Reply:
I've updated the file name: 592480d714f9d4ff112239a2028f5c6ba146ed99